File: data/mi1/rooms/items.lua
-- generally, an item is a blueprint for an object
-- * it can have graphics attached. To do so, model needs to be specified
engine.items["guybrush"] = {
	tag = "player",
	model = "guybrush",
	character = {
		state = "idle",
		speed = 500
	},
	text_color = {255, 255, 255, 255},
	text_offset = {0, 60},
	applydepth = true

}

engine.items["pieces_of_eight"] = {
	hotspot = {
		text = strings.objects.pieceofeight,
		text_plural = strings.objects.piecesofeight,
	},
	actions = {
		look = {
			{ type = scumm.action.say, args= { actor="guybrush", lines = {strings.pieces_of_eight}}}
		}

 -- function () 
	-- 		print ("123")		
	-- 		local s = script:new()
	-- 		s.actions = {
	-- 			action.say {id=1, actor="guybrush", lines = {strings.pieces_of_eight()}}
	-- 		}
	-- 		return s
	-- 	end

	}
}

engine.items["map"] = {
	hotspot = {
		text = strings.objects.map
	},
	actions = {
		look = function()
			local act = {
				{ type = action.animate, args = {tag="player", anim="open_map", sync=true}}
			}
			if (variables.map_looked == false) then 
				variables.map_looked = true
				table.insert (act, {type=action.delay, args = {sec=1}})
				table.insert (act, {
					type = scumm.action.say, args = {actor="guybrush", animstart="open_map_talk", animend="open_map_idle", lines = 
						{strings.village2[4], strings.village2[5], strings.village2[6], strings.village2[7]}}
				})
			else
				table.insert(act, {type=action.animate, args ={tag="player", anim="open_map_idle"}})
			end
				table.insert (act, {
					type = action.callfunc, args = { func = function() 
						local p = monkey.getEntity("main")
						local ui = monkey.getEntity("ui")
						local ot = monkey.getEntity("other")
						monkey.enablescriptengine(false)
   						p:setactive(false)
   						ui:setactive(false)
						local id = monkey.addEntity ({pos={0,0,0}, components = {
							{type="gfx", image="map.png"}
						}},ot)
						engine.state.scumm.lmboverride = function()
							ot:clear()
							p:setactive(true)
							ui:setactive(true)
							monkey.enablescriptengine(true) 
							local act2 = {
								{ type=action.animate, args = {tag="player", anim="open_map", sync=true, fwd=false}},
								{ type=action.animate, args = {tag="player", anim="idle_e"}}
							}
							local s = script.make(act2)
							engine.state.scumm.lmboverride=nil
							monkey.play(s)
						end
					end }
				})
			return act
		end
	}
}

File: data/mi1_py/data/items.py

from lib_py.scumm.scumm import State
import lib_py.engine as engine
#from lib_py.scumm.scumm import DynamicItem
import lib_py.scumm.entity as se
import lib_py.components as compo

st = engine.data['strings']

# guybrush
State.addItem (se.CharItem (
    id = 'guybrush',
    model = 'guybrush',
    speed = 500,
    text_color = [255, 255, 255, 255],
	text_offset = [0, 60]))

State.addItem(se.Item(
    id = 'pieces.of.eight',
    text = st['objects']['pieceofeight'],
    plural= st['objects']['piecesofeight']
))

# engine.addEntity(
#     id = 'guybrush',
#     e = se.Character(
#         model = 'guybrush', 
#         speed = 100, 
#         dir = 'e', 
#         state = 'idle',
#         text_color = [255, 255, 255, 255],
# 	    text_offset = [0, 60]
#     ))

# make guybrush!
# def makeGuybrush(**kwargs):
#     print (kwargs)
#     state = kwargs['state'] if 'state' in kwargs else 'idle'
#     dir = kwargs['dir'] if 'dir' in kwargs else 'e'
#     e = se.Character (model='guybrush', speed=100, dir=dir, state=state, text_color = [255,255,255,255], text_offset=[0,60])
#     if 'pos' in kwargs:
#         e.pos = kwargs['pos']
#         print ('found pos ' + str(e.pos[0]))
#     if State.player == 'guybrush':
#         e.tag = 'player'
#         e.addComponent (compo.Follow())
#     else:
#         e.tag = 'guybrush'
#     return e

#engine.data['factories']['guybrush'] = makeGuybrush

# a dynamic item is created ON THE FLY. It is simply a callable that creates an entity
